core/src/main/java/org/testcontainers/containers/ContainerState.java

```diff
@@ -319,4 +319,23 @@ default void copyFileFromContainer(String containerPath, String destinationPath)
             return function.apply(tarInputStream);
         }
     }
+
+    class InternalMethods {
```

Comment:
this class is public (as any other member of an interface). Please make it at least package internal.

Reply:
you literally said:
> One way to do it is to define the method in an internal class.

It's not possible to make an inner class on an interface anything other than public

Reply:
I did not say that you should make an inner class of an interface. There are many ways of defining internal classes.